### PR TITLE
update memory monitor

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Updates memory monitor to use GB, which is much more readable. Also fixes it to track current allocated memory instead of cumulative (which isn't really useful)

# What issue(s) does this change relate to?

[CO-1740](https://mosaicml.atlassian.net/browse/CO-1740)

[CO-1740]: https://mosaicml.atlassian.net/browse/CO-1740?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: composer/callbacks/memory_monitor.py
Comment (dakinggg) [inline composer/callbacks/memory_monitor.py:118]: should this be rounded to some number of decimals?

**Comment by bandish-shah** [comment]
Could we post an example of the new output before we're ready to merge this PR in the description?

**Review by dakinggg — CHANGES_REQUESTED** [review]
Blocking to look at the output and approve

**Comment by mvpatel2000** [comment]
image

**Review by dakinggg — COMMENTED** [review]
Could you also paste in what this looks like when you log to console? Otherwise LGTM and will approve after that.

**Comment by mvpatel2000** [comment]
```[epoch=1][batch=96/390]:
	 Train trainer/global_step: 95
	 Train trainer/batch_idx: 95
	 Train memory/allocated_mem: 22.9840
	 Train memory/active_mem: 22.9840
	 Train memory/inactive_mem: 5.3172
	 Train memory/reserved_mem: 38.9250
	 Train memory/alloc_retries: 0
	 Train trainer/device_train_microbatch_size: 10
	 Train loss/train/total: 0.1528
	 Train wall_clock/train: 627.0995
	 Train wall_clock/val: 0.0000
	 Train wall_clock/total: 627.0995```

**Pull request merged**
